src/schema/rules/sidecars/asl_validation.yaml

```diff
@@ -0,0 +1,259 @@
+        zero.
+        In case an array of numbers is provided, its length should be equal to the number of volumes specified in
+        `*_aslcontext.tsv`. Corresponds to DICOM Tag 0018,9258 `ASL Pulse Train Duration`.
+    level: error
+    selectors:
+    - suffix == "asl"
+    - sidecar contains "LabelingDuration"
+    - type(sidecar.LabelingDuration) == "seq"
+    checks:
+    - nifti_header.pixdim[4] == sidecar.LabelingDuration.size
+
+# 165
+ASLContextConsistent:
+    code: ASLCONTEXT_TSV_NOT_CONSISTENT
+    description: |
+        The number of volumes in the '*_aslcontext.tsv' for this file does not match the number of
+        values in the NIfTI header.
+    level: error
+    selectors:
+    - suffix == "asl"
+    - associated contains "aslcontext"
+    checks:
+    - nifti_header.pixdim[4] == aslcontext.n_rows
+
+# 168
+ASLFlipAngleNiftiLength:
+    code: FLIP_ANGLE_NOT_MATCHING_NIFTI
+    description: |
+        The number of values for 'FlipAngle' for this file does not match the 4th dimension of the NIfTI header.
+        'FlipAngle' is the flip angle (FA) for the acquisition, specified in degrees.
+        Corresponds to: DICOM Tag 0018, 1314 `Flip Angle`.
+        The data type number may apply to files from any MRI modality concerned with a single value for this field,
+        or to the files in a file collection where the value of this field is iterated using the flip entity.
+        The data type array provides a value for each volume in a 4D dataset and should only be used when the
+        volume timing is critical for interpretation of the data, such as in ASL or variable flip angle fMRI sequences.
+    level: error
+    selectors:
+    - suffix == "asl"
+    - sidecar contains "FlipAngle"
+    - type(sidecar.FlipAngle) == "seq"
+    checks:
+    - nifti_header.pixdim[4] == sidecar.FlipAngle.size
+
+# 172
+ASLFlipAngleASLContextLength:
+    code: FLIP_ANGLE_NOT_MATCHING_ASLCONTEXT_TSV
+    description: |
+       The number of values for 'FlipAngle' for this file does not match the number of volumes in the
+       'sub-<label>[_ses-<label>][_acq-<label>][_rec-<label>][_run-<index>]_aslcontext.tsv'.
+       'FlipAngle' is the flip angle (FA) for the acquisition, specified in degrees.
+       Corresponds to: DICOM Tag 0018, 1314 `Flip Angle`.
+       The data type number may apply to files from any MRI modality concerned with a single value for this field,
+       or to the files in a file collection where the value of this field is iterated using the flip entity.
+       The data type array provides a value for each volume in a 4D dataset and should only be used when the volume
+       timing is critical for interpretation of the data, such as in ASL or variable flip angle fMRI sequences.
+    level: error
+    selectors:
+    - suffix == "asl"
+    - associated contains "aslcontext"
+    - sidecar contains "FlipAngle"
+    - type(sidecar.FlipAngle) == "seq"
+    checks:
+    - aslcontext.n_rows == sidecar.FlipAngle.size
```

Comment:
Redundant.

```suggestion
```

Reply:
WDYT of implementing any rules we can in one PR to master, and then identifying and removing redundant ones in a separate one? Just so that our decision to remove unnecessary rules is easily traceable.